### PR TITLE
Fix even more 64-bit warnings

### PR DESCRIPTION
**Pull request by asarium**

### CODE REVIEW

**Comment by MageKing17** [comment]
Nothing jumps out at me from this batch.

**Comment by asarium** [comment]
I fixed a few GCC warnings introduces by this commit and I also fixed some existing warnings. The last commit is a bit special since I converted the fix conversion macros to `constexpr` function to get rid of the negative shift warning.

File: code/network/multiui.cpp [hunk 1]
```diff
@@ -4977,6 +4977,7 @@ void multi_create_accept_hit()
 	// coop game option validation checks
 	if ( (Netgame.type_flags & NG_TYPE_COOP) && (Net_player->flags & NETINFO_FLAG_AM_MASTER) ) {
 		// check for time limit
+
```
Comment (MageKing17) [inline code/network/multiui.cpp]: Spurious newline insertion?

**Comment by asarium** [comment]
Accidentally added newline has been removed.

**Pull request merged**